### PR TITLE
Feature/ui workflow babelify

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds babel to gulp build process
- Adds sourcemaps to debug es6 code in browsers
- Modifies workflows feature to use basic es6 features.

es6workflows
Cluster: nodeproxy1736-1000.dev.continuuity.net:9999

### CODE REVIEW

File: cdap-ui/app/features/workflows/controllers/tabs/schedule-ctrl.js [hunk 1]
```diff
@@ -1,87 +1,92 @@
...
+      scope: this.$scope
+    }, {},
+    function success() {},
+    function error(err) {
+      this.myAlert({
+        title: 'Cannot Suspend Schedule',
+        content: err
       });
-    };
+    }.bind(this));
```
Comment (elfenheart) [inline cdap-ui/app/features/workflows/controllers/tabs/schedule-ctrl.js]: do we still need the .bind(this)?

File: cdap-ui/app/features/workflows/controllers/tabs/schedule-ctrl.js [hunk 2]
```diff
@@ -1,87 +1,92 @@
...
+      scheduleId: obj.schedule.name,
+      scope: this.$scope
+    }, {},
+    function success() {},
+    function error(err) {
+      this.myAlert({
+        title: 'Cannot Resume Schedule',
+        content: err
+      });
+    }.bind(this));
```
Comment (elfenheart) [inline cdap-ui/app/features/workflows/controllers/tabs/schedule-ctrl.js]: same comment as line 72

File: cdap-ui/app/features/workflows/service/graph-helpers.js [hunk 3]
```diff
@@ -118,9 +119,33 @@ angular.module(PKG.name + '.services')
       }
     }
 
+    function addStartAndEnd(nodes) {
```
Comment (elfenheart) [inline cdap-ui/app/features/workflows/service/graph-helpers.js]: do we need the *function* keyword?

File: cdap-ui/app/features/workflows/controllers/tabs/schedule-ctrl.js (hunk 1 above)
Comment (ajainarayanan) [inline cdap-ui/app/features/workflows/controllers/tabs/schedule-ctrl.js]: Only fat arrow functions ```()=>``` don't need the executing scope. Apart from that if we are using ```function () {..}``` syntax we need to bind it with a scope if it needs to be executed in a specific scope.

File: cdap-ui/app/features/workflows/controllers/tabs/schedule-ctrl.js (hunk 2 above)
Comment (ajainarayanan) [inline cdap-ui/app/features/workflows/controllers/tabs/schedule-ctrl.js]: Only fat arrow functions ```()=>``` don't need the executing scope. Apart from that if we are using ```function () {..}``` syntax we need to bind it with a scope if it needs to be executed in a specific scope.

File: cdap-ui/app/features/workflows/service/graph-helpers.js (hunk 3 above)
Comment (ajainarayanan) [inline cdap-ui/app/features/workflows/service/graph-helpers.js]: function still exists in es6. We just have an extra syntax (fat arrow functions) that allows us to express anonymous functions.

**Comment by ajainarayanan** [comment]
TODO: We need to modify our gulp build process for development. Right now each change takes around 7s to build app.js file and reload the browser. Its extremely in-efficient for us during development. Will fix it in the next PR.

**Comment by elfenheart** [comment]
@ajainarayanan I think part of modifying the Gulp should be deprecating angular-strap modal (or if we can even remove angular-strap completely).

We have to modify Cask Confirmable for that though...

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**